### PR TITLE
[EPROD-533] Minter canister API and client

### PR DESCRIPTION
**Pull request by F3kilo**
# Issue ticket

Issue ticket link:
https://infinityswap.atlassian.net/browse/EPROD-533

### Code conventions

- [X] I have performed a self-review of my code
- [X] Every new function is documented
- [X] Object names are auto explicative

### Security

- [X] The PR does not break APIs backward compatibility
- [X] The PR does not break the stable storage backward compatibility

### Testing

- [X] Every function is properly unit tested
- [X] I have added integration tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] IC endpoints are always tested through the `canister_call!` macro

### CODE REVIEW

File: src/minter-client/scripts/build_test_canister.sh
Comment (ufoscout) [inline src/minter-client/scripts/build_test_canister.sh:1]: should this be integrated with the CI?

File: src/minter-api/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+[package]
+categories = ["cryptography::cryptocurrencies"]
+description = "API types definition for minter canister"
+include = ["src/**/*", "../../LICENSE", "../../README.md"]
+name = "minter-api"
```
Comment (ufoscout) [inline src/minter-api/Cargo.toml:5]: I guess this should be called `minter-did`

**Review by ufoscout — CHANGES_REQUESTED** [review]

File: src/minter-client/scripts/build_test_canister.sh
Comment (F3kilo) [reply src/minter-client/scripts/build_test_canister.sh:1]: Oh, it is copy/paste error. I forgot to remove this tests for another package.

**Review by GraDKh — APPROVED** [review]

File: src/minter-client/scripts/build_test_canister.sh
Comment (ufoscout) [inline src/minter-client/scripts/build_test_canister.sh:1]: how is this file linked to the minter client?

File: src/minter-client/scripts/dfx_deploy.sh
Comment (ufoscout) [inline src/minter-client/scripts/dfx_deploy.sh:1]: how is this file linked to the minter client?

File: src/minter-client/scripts/download_iceth.sh
Comment (ufoscout) [inline src/minter-client/scripts/download_iceth.sh:1]: how is this file linked to the minter client?

**Review by ufoscout — CHANGES_REQUESTED** [review]

File: src/minter-client/scripts/build_test_canister.sh
Comment (F3kilo) [reply src/minter-client/scripts/build_test_canister.sh:1]: Removed

File: src/minter-client/scripts/dfx_deploy.sh
Comment (F3kilo) [reply src/minter-client/scripts/dfx_deploy.sh:1]: Removed

File: src/minter-client/scripts/download_iceth.sh
Comment (F3kilo) [reply src/minter-client/scripts/download_iceth.sh:1]: Removed

**Review by ufoscout — APPROVED** [review]

**Pull request merged**